### PR TITLE
Feat/new docs

### PR DESCRIPTION
**Pull request by felippepuhle**
Closes https://github.com/wcandillon/react-native-redash/issues/156

This is a first step to unify the docs to this repo.

We have a lot of things to discuss here:

- Is this the best approach?
- If yes, will we create a separate package for the lib itself? (this would allow us to import it locally instead of adding react-native-redash as a dependency)
- Standardize code/config/linters (gatsby uses prettier, maybe it's a good idea to add it to the whole project)
- Layout can be improved, I'm using random colors as theme and we don't have a logo/favicon
- Search isn't working yet
- Will we keep the docs on README.md? 

A simple `yarn docs deploy` should put it online under https://wcandillon.github.io/react-native-redash

Demo: https://felippepuhle.github.io/react-native-redash

If you guys agree I think we can go ahead and improve those things later on.

Thanks

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +9,9 @@
     "tsc": "tsc --noEmit",
     "test": "jest",
     "prepare": "bob build",
-    "semantic-release": "semantic-release"
+    "semantic-release": "semantic-release",
+    "docs:start": "yarn workspace docs run start",
```
Comment (dsznajder) [inline package.json]: AFAIK, You don't need `run` on `yarn`.
Also, I'm thinking if we need 2 scripts for that, maybe just aliasing workspace is enough.

```suggestion
    "docs": "yarn --cwd docs",
```

Comment (felippepuhle) [reply package.json]: `yarn --cwd docs` doesn't work for that (just tested id), if we want to use `yarn docs start` we must use `"docs": "yarn workspace docs"`

Comment (felippepuhle) [reply package.json]: Fixed with `"yarn workspace docs"`! thanks

**Comment by wcandillon** [comment]
@felippepuhle This is amazing thank you. Better documentation has been by far the most requested feature on redash. I have a couple of questions since I am completely new to Gatsby.

What is the editing workflow? We modify the JavaScript files directly? I've ran the build command successfully. But where are the build file? I didn't find them.

How was the doc created? Is this typical gatsby app with a special theme for the documentation?

Thank you for your contribution.

**Comment by wcandillon** [comment]
@felippepuhle I remember that you told me about docz, did you decide to go to a different route?

**Comment by felippepuhle** [comment]
Hey @wcandillon!

There are actually a lot of solutions and Docs is just one of them. But under the hoods, Docz uses Gatsbsy and I wanted to stay as pure as possible.

About the workflow, you cant start the dev server by just typing `yarn docs start` at the root folder.
You can create a fresh build by typing `yarn docs build`, it should create the files under `./packages/docs/public`.
For deploy purpose you can use `yarn docs deploy`, it'll create a production build and send to `gh-pages` branch.

I didn't  change anything on the CI yet, but I was thinking that maybe we could run `yarn docs deploy` [here](https://github.com/wcandillon/react-native-redash/blob/master/.circleci/config.yml#L44).

Thanks!

File: packages/docs/src/pages/math/clamp.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react"
+
+import { CodeHighlight, SEO } from "src/components"
+
+const ClampPage: React.FC = () => (
+  <>
+    <SEO title="clamp - Math" />
+
+    <h1>clamp(node: Node, lowerBound: Adaptable, upperBound: Adaptable)</h1>
```
Comment (wcandillon) [inline packages/docs/src/pages/math/clamp.tsx]: This is where I would edit the documentation right?

File: package.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "react-native-redash",
+  "private": true,
```
Comment (wcandillon) [inline package.json]: Won't this prevent me from publishing? Do we have to move redash in `/packages`?

**Comment by wcandillon** [comment]
A huge thank you @felippepuhle. Still, a lot of work to be done documentation-wise but you've unlocked it for us :) Thank you!

**Pull request closed without merging**